script.module.libmediathek3/lib/libmediathek3dialogs.py

```diff
@@ -19,10 +19,6 @@ def dialogDate():
 	return dialog.numeric(1, libmediathek3utils.getTranslation(31030)).replace('/','').replace(' ','0')
 	
 def getSearchString():
-	if libmediathek3utils.searchWorkaroundExists():
-		d = libmediathek3utils.searchWorkaroundRead()
-	else:
-		dialog = xbmcgui.Dialog()
-		d = dialog.input(libmediathek3utils.getTranslation(31039),type=xbmcgui.INPUT_ALPHANUM)
-		libmediathek3utils.searchWorkaroundWrite(d)
+	dialog = xbmcgui.Dialog()
+	d = dialog.input(libmediathek3utils.getTranslation(31039),type=xbmcgui.INPUT_ALPHANUM)
```

Comment:
If I am not mistaken, d will return -1 if canceled. Are you taking this into account?